proposals/12020-model-registry-integration/README.md

```diff
@@ -0,0 +1,432 @@
+    model: dsl.Output[dsl.Model],
+):
+    # Training logic
+    with open(model.path, "r") as model_file:
+        print("Training the model...")
+
+    # Set model metadata
+    model.name = "my-model-v1.0.0"
+    model.metadata["training_epochs"] = 100
+    model.metadata["accuracy"] = 0.95
+
+    # Register model with Model Registry
+    model.register(
+        model_name="sentiment-classifier",
+        description="BERT-based sentiment classification model",
+        model_format_name="vLLM",
+        model_format_version=None,
+        owner="ml-team",
+        author="data-scientist@company.com",
+        continue_on_error=True,  # Default: True
+    )
+```
+
+#### API Reference
+
+The `model.register()` method accepts the following parameters:
+
+| Parameter              | Type | Required | Default              | Description                                          |
+| ---------------------- | ---- | -------- | -------------------- | ---------------------------------------------------- |
+| `model_name`           | str  | Yes      | -                    | Name of the model in Model Registry                  |
+| `description`          | str  | No       | ""                   | Human-readable model description                     |
+| `model_format_name`    | str  | No       | None                 | Model format (e.g., "PyTorch", "TensorFlow", "ONNX") |
+| `model_format_version` | str  | No       | None                 | Version of the model format                          |
+| `owner`                | str  | No       | "Kubeflow Pipelines" | Model owner/team                                     |
+| `author`               | str  | No       | "Kubeflow Pipelines" | Model author                                         |
+| `continue_on_error`    | bool | No       | True                 | Whether to fail pipeline on registration error       |
+
+#### Backend Translation
+
+The SDK call translates to the following Model Registry API invocation:
+
+```python
+# Equivalent Model Registry client call
+registered_model = registry.register_model(
+    name="sentiment-classifier",
+    description="BERT-based sentiment classification model",
+    owner="ml-team",
+    author="data-scientist@company.com",
+    version="my-model-v1.0.0",  # model.name
+    uri="s3://kfp-artifacts/run-123/model",  # model.uri
+    metadata={
+        "training_epochs": 100,
+        "accuracy": 0.95
+    },  # model.metadata
+    model_format_name="vLLM",
+    model_format_version=None,
+    model_source_id="b6a9dde3-1647-463f-aeb8-5800089c84e8",  # Pipeline run ID
+    model_source_name="sentiment-training-pipeline",  # Pipeline run name
+    model_source_class="pipelinerun",  # KFP-specific identifier
+    model_source_kind="kfp",  # KFP-specific identifier
+    model_source_group="ml-team",  # Pipeline namespace
+)
+```
+
+#### Cross-Reference Metadata
+
+After successful registration, KFP adds metadata to the model artifact for UI cross-referencing:
```

Comment:
this will be very helpful to go DSP -> MR, in addition to the opposite direction thanks to the `model_source*` fields above.

Reply:
Yes, exactly. It's modeled after the InstructLab pipeline so with minor tweaks to the Model Registry dashboard, we should get really nice linking.